### PR TITLE
DOC: fix RT03,SA01,ES01 for pandas.core.resample.Resampler.__iter__

### PR DESCRIPTION
**Pull request by tuhinsharma121**
fixes

```
pandas.core.resample.Resampler.__iter__ RT03,SA01,ES01
pandas.core.groupby.DataFrameGroupBy.__iter__ RT03,SA01
pandas.core.groupby.SeriesGroupBy.__iter__ RT03,SA01
```

### CODE REVIEW

**Review by rhshadrach — APPROVED** [review]
lgtm

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -767,10 +767,23 @@ def __iter__(self) -> Iterator[tuple[Hashable, NDFrameT]]:
         """
         Groupby iterator.
 
+        This method provides an iterator over the groups created by the resampling
+        operation on the object. The method yields tuples where the first element
+        is the label (group key) corresponding to each resampled bin, and the second
+        element is the subset of the data that falls within that bin.
```
Comment (rhshadrach) [inline pandas/core/groupby/groupby.py:773]: This docstring is shared by groupby and resample, can you write this in a way that covers both of these cases?

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (tuhinsharma121) [reply pandas/core/groupby/groupby.py:773]: yes. on it.

Comment (tuhinsharma121) [reply pandas/core/groupby/groupby.py]: done

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -767,10 +767,24 @@ def __iter__(self) -> Iterator[tuple[Hashable, NDFrameT]]:
         """
         Groupby iterator.
 
+        This method provides an iterator over the groups created by the ``resample``
+        or ``groupby`` operation on the object. The method yields tuples where
+        the first element is the label (group key) corresponding to each resampled
+        bin, and the second element is the subset of the data that falls within that
+        bin.
```
Comment (rhshadrach) [inline pandas/core/groupby/groupby.py:774]: `corresponding to each group or resampled bin` and `falls within that group or bin`.

File: pandas/core/groupby/groupby.py [hunk 3]
```diff
@@ -767,10 +767,24 @@ def __iter__(self) -> Iterator[tuple[Hashable, NDFrameT]]:
...
+
         Returns
         -------
-        Generator yielding sequence of (name, subsetted object)
-        for each group
+        Iterator
+            Generator yielding a sequence of (name, subsetted object)
+            for each resampled group.
```
Comment (rhshadrach) [inline pandas/core/groupby/groupby.py:780]: I would recommend just keeping this as `group`.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/groupby.py (hunk 2 above)
Comment (tuhinsharma121) [reply pandas/core/groupby/groupby.py]: apologies for missing this. fixed now.

File: pandas/core/groupby/groupby.py (hunk 3 above)
Comment (tuhinsharma121) [reply pandas/core/groupby/groupby.py]: ixed now.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Pull request merged**